### PR TITLE
Move the environment service to its own TLJH plugin

### LINKED ISSUE #136: Move the TLJH docker plugin to its own repo
For now there is a single TLJH plugin for the whole PlasmaBio stack

However, some users might be interested in using the "Environments" part only. That means being able to build local Docker images from the JupyterHub interface, but without the opinionated choice of using `PAMAuthenticator` and `SystemUserSpawner`. For example they could use it on a temporary instance to build a few Docker images and start them with `DockerSpawner`, which using GitHub for authentication.

PlasmaBio would consume this TLJH plugin either as a Python dependency, or by being installed side-by-side (with an extra `--plugin` parameter in the TLJH installer command).

### PR DESCRIPTION
**Pull request by jtpio**
Fixes #136.

Move the building of the Docker images to its own TLJH plugin, so it's easier to reuse in other TLJH deployments that don't use the same defaults as PlasmaBio.

The plugin is available at: https://github.com/plasmabio/tljh-repo2docker

- [x] Add `tljh-repo2docker` to the plugin dependencies
- [x] Update the documentation to mention the plugin

### CODE REVIEW

**Comment by pierrepo** [comment]
@jtpio just to be sure, the `tljh-plasmabio` *local* plugins becomes `tljh-repo2docker`?

**Comment by pierrepo** [comment]
Do you know if it's possible to move issues from a repo to another?

I'm thinking to those ones:
- #142 
- #60
- #46

**Comment by jtpio** [comment]
> just to be sure, the `tljh-plasmabio` _local_ plugins becomes `tljh-repo2docker`

`tljh-plamasbio` will still exist, but will be more about setting the JupyterHub config such as the PAMAuthenticator and using system users. And will also be much smaller in scope.

So the PlasmaBio would depend on both `tljh-repo2docker` and `tljh-plasmabio`.

> Do you know if it's possible to move issues from a repo to another?

Yes indeed, this can be done when this PR is complete using the Transfer issue button on the issue page:

image

**Comment by pierrepo** [comment]
>So the PlasmaBio would depend on both tljh-repo2docker and tljh-plasmabio.

Perfect!

>when this PR is complete using the Transfer issue button

Good. And I guess the whole discussion is also transferred.

**Comment by jtpio** [comment]
> And I guess the whole discussion is also transferred.

Yes and there is even some status message that indicates that the issue was moved.

File: ansible/tljh.yml [hunk 1]
```diff
@@ -24,18 +24,25 @@
...
       environment:
         TLJH_BOOTSTRAP_PIP_SPEC: "{{ tljh_bootstrap_pip_spec }}"
 
     - name: Set the idle culler timeout to 1 hour
       shell: "tljh-config set services.cull.timeout 3600"
 
+    - name: Set the default memory and cpu limits
+      shell: |
+        tljh-config set limits.memory 2G
+        tljh-config set limits.cpu 2
```
Comment (jtpio) [inline ansible/tljh.yml:44]: We can now set the default limits using `tljh-config`, which is better than having them hardcoded in the Python code of the plugin.

The `tljh-repo2docker` plugin is then able to pick them up to display the defaults in the UI: https://github.com/plasmabio/tljh-repo2docker/blob/ec9e2dee20095b11b4a933915694178775d50611/tljh_repo2docker/__init__.py#L80-L84

**Comment by jtpio** [comment]
Tested on a fresh VM and the combination of the 2 plugins seems to be working as expected.

We can consider moving some of the documentation to the `tljh-repo2docker` repo in a separate change, as mentioned in: https://github.com/plasmabio/tljh-repo2docker/issues/4

**Pull request merged**